File: webapp.client/src/app/Dtos/PageDto.ts
export interface Page {
  id: string;
  title: string;
  userId: string;
  parentId?: string;
  createdAt: string;
  updatedAt: string;
  pageSections: PageSection[];
}

export interface PageSection {
  id: string;
  pageId: string;
  order: number;
  createdAt: string;
  pageContents: PageContent[];
}

export interface PageContent {
  id: string;
  sectionId: string;
  contentType: string;
  contentData: string;
  createdAt: string;
}

export interface PageDto {
  title: string;
  userId: string;
  parentId?: string | null; // Modifica per consentire 'null' oltre a 'string'
  pageSections: PageSectionDto[];
}


export interface PageSectionDto {
  order: number;
  pageContents: PageContentDto[];
}

export interface PageContentDto {
  contentType: string;
  contentData: string;
}

File: BusinessLogic/Dtos/PageDto.cs
namespace BusinessLogic.Dtos
{
    public class PageDto
    {
        public Guid Id { get; set; }
        public required string Title { get; set; }
        public Guid UserId { get; set; }
        public Guid? ParentId { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public List<PageSectionDto> PageSections { get; set; } = new();
    }

    public class PageSectionDto
    {
        public Guid Id { get; set; }
        public Guid PageId { get; set; }
        public int Order { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public List<PageContentGroupDto> PageContentGroups { get; set; } = new();
    }

    public class PageContentGroupDto
    {
        public Guid Id { get; set; }
        public Guid SectionId { get; set; } // Riferisce a PageSection
        public int Order { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public List<PageContentDto> PageContents { get; set; } = new();
    }

    public class PageContentDto
    {
        public Guid Id { get; set; }
        public Guid ContentGroupId { get; set; }  // Riferisce a PageContentGroup
        public string? ContentType { get; set; }
        public string? ContentData { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public int Order { get; set; } // Ordine all'interno del ContentGroup
    }
}
